tabs/pid_tuning.js

```diff
@@ -701,11 +718,8 @@ TABS.pid_tuning.initRatesPreview = function () {
     this.keepRendering = true;
     this.model = new Model($('.rates_preview'), $('.rates_preview canvas'));
 
-    var scale = d3.scale.linear().domain([900, 2100]);
-
-    this.rollScale = scale.range([Math.PI * 2, -Math.PI * 2]);
-    this.pitchScale = scale.range([Math.PI * 2, -Math.PI * 2]);
-    this.yawScale = scale.range([Math.PI * 2, -Math.PI * 2]);
+    this.$superExpo = $('.pid_tuning input[name="show_superexpo_rates"]');
```

Comment:
I don't seem to be seeing closures as such a bad thing as you seem to be seeing them.

The 'private by convention' method is a viable approach - but as this project does not yet have a set of coding style rules, unfortunately this is not really an option. Combine this with the more or less total absence of code reviewing in this project, and this looks like a recipe for code to break.

Once we have put these things in place, scoping by convention is certainly a possibility.

I don't like the use of `private` helper objects. In a non-flat model, this quickly becomes clumsy, e.g. `this.foo.private.bar.private.baz`.

Re your remark about dev tools: Do you have any recommendations for a good free IDE for JS that runs under linux? My dev tools currently are vim and the Chrome console.

Agreed about patterns, and also agreed that the current use of closures is a mess. I would be willing to allow non-nested closures as one way of encapsulation, though.

I have rewritten my rates curves to make use of `currentRates` in #96, and I have also tried to flatten out the closures as far as possible without rewriting the whole thing. Also, I have added some code to your way of updating `currentRates` to prevent it from breaking with non-current versions of the firmware.

One thing that I noticed re your coding style is that you are using a mix of CamelCase and underscore.  I am aware that the current project is a mess with respect to this, but doing this inside one object `currentRates.roll_rate` is not exactly helpful to improve this. ;-)